server/passport.ts

```diff
@@ -318,6 +343,21 @@ function setupPassport(models) {
         location: profile._json.location,
       }
     });
+
+    // Handle OAuth for custom domains.
+    //
+    // If req.query.from is a valid custom domain for a community,
+    // associate our LoginToken with this Github account. We will
+    // redirect to [customdomain] afterwards and consume this
+    // LoginToken to get a new login session.
+    if ((req as any).loginTokenForRedirect) {
```

Comment:
Duplicate code from above -- should we refactor into a function?

Reply:
I'm okay with leaving this in - the logic is pretty straightforward, and the comment makes it look more significant than it otherwise is (the branch between githubAccount and newGithubAccount inherently repeats a lot of code, but it seems better than trying to collapse the branch which would be logically confusing).